controllers/Backend.php

```diff
@@ -1,104 +1,351 @@
+                $errors['content'] = "Votre article doit contenir au moins 100 caractères.";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable";
+                }
+            }
+
+            if(empty($errors)){
+                $manager_article->post($title,$chapo,$content,$posted);
+                if(!empty($_FILES['image']['name'])){
+                    $manager_article->post_img($_FILES['image']['tmp_name'], $extension);
+                    header("Location:/liste-de-tous-les-articles");
+                }else{
+                   header("Location:/liste-de-tous-les-articles");
+                }
+            }
+        }
+        else
+            $errors =[];
+   
+        if(!isset($_SESSION['admin'])){
+            header("location: /articles-a-la-une");
+        }
+        
+        $myView = new View('write', 'backend');
+        $myView->render(array('errors' => $errors));
     }
 
     public function settings()
     {
-        $model_user = new User();
+        $manager_user = new UserManager();
+        $modos = $manager_user->get_modos();
+ 
+        if(filter_has_var(INPUT_POST, 'submit')){
+                if(filter_has_var(INPUT_POST, 'name')){
+                    $name = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'name')));
+                }
+                if(filter_has_var(INPUT_POST, 'email')){
+                    $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email')));
+                }
+                if(filter_has_var(INPUT_POST, 'email_again')){
+                    $email_again = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email_again', FILTER_VALIDATE_EMAIL)));
+                }
+                if(filter_has_var(INPUT_POST, 'role')){
+                    $role = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'role')));
+                }
+                
+                $token = $manager_user->token(30);
+
+                $errors = [];
+
+                if(empty($name) || empty($email) || empty($email_again)){
+                    $errors['empty'] = "Veuillez remplier tous les champs";
+                }
+
+                if($email != $email_again){
+                    $errors['different'] = "Les adresses email ne correspondent pas";
+                }
+
+                if($manager_user->email_taken($email)){
+                    $errors['taken'] = "L'adresse email est déjà assignée à un modérateur";
+                }
+
+                if(empty($errors)){
+                    $manager_user->add_modo($name,$email,$role,$token);
+                    header("Location:/gestion-des-admins-et-modos");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function header() is discouraged](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)